File: VB/SelectPrinterInCode/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports System.Drawing.Printing
Imports DevExpress.XtraPrinting
' ...

Namespace SelectPrinterInCode
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim rep As New XtraReport1()
			AddHandler rep.PrintingSystem.StartPrint, AddressOf printingSystem_StartPrint
			rep.Print()
		End Sub

		Private Sub printingSystem_StartPrint(ByVal sender As Object, ByVal e As PrintDocumentEventArgs)
			' set specific printer name
			e.PrintDocument.PrinterSettings.PrinterName = PrinterSettings.InstalledPrinters(0)
		End Sub

	End Class
End Namespace

File: CS/SelectPrinterInCode/Form1.cs
using System;
using System.Windows.Forms;
using System.Drawing.Printing;
using DevExpress.XtraPrinting;
// ...

namespace SelectPrinterInCode {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 rep = new XtraReport1();
            rep.PrintingSystem.StartPrint +=
                new PrintDocumentEventHandler(printingSystem_StartPrint);
            rep.Print();
        }

        private void printingSystem_StartPrint(object sender, PrintDocumentEventArgs e) {
            // set specific printer name
            e.PrintDocument.PrinterSettings.PrinterName =
                PrinterSettings.InstalledPrinters[0];
        }

    }
}